enclave-runtime/src/lib.rs

```diff
@@ -308,18 +309,20 @@ pub unsafe extern "C" fn init_parentchain_components(
 	let mut params = slice::from_raw_parts(params, params_size);
 	let latest_header_slice = slice::from_raw_parts_mut(latest_header, latest_header_size);
 
-	let params = match LightClientInitParams::<Header>::decode(&mut params) {
+	let params = match ParentchainInitParams::decode(&mut params) {
```

Comment:
This part of the enclave should not need to know which type of header is used, this should be the task of the specific parentchain initialization function. Therefore, it was removed, and an encoded header is returned instead (see change below).